lib/rucio/daemons/replicarecoverer/suspicious_replica_recoverer.py

```diff
@@ -447,15 +425,14 @@ def run(once=False, younger_than=3, nattempts=10, vos=None, limit_suspicious_fil
     max_offset = timedelta(hours=1, seconds=10)
     if isinstance(db_time, datetime):
         if db_time - client_time > max_offset or client_time - db_time > max_offset:
-            logger(logging.CRITICAL, 'Offset between client and db time too big. Stopping Suspicious-Replica-Recoverer.')
+            # logger(logging.CRITICAL, 'Offset between client and db time too big. Stopping Suspicious-Replica-Recoverer.')
+            logging.critical('Offset between client and db time too big. Stopping Suspicious-Replica-Recoverer.')
             return
 
-    sanity_check(executable='rucio-replica-recoverer', hostname=socket.gethostname())
-
     if once:
```

Comment:
I have the impression that this is unnecessary and can be handled by `run_daemon()` instead.

Reply:
Sorry, I now realise my comment can be confusing. I’m referring to:

https://github.com/rucio/rucio/blob/76f3e26df77f822f071c640aab9e422e4b77baee/lib/rucio/daemons/replicarecoverer/suspicious_replica_recoverer.py#L431-L448

Is there a benefit to checking `once` in this part of the code? `run_daemon` can also ensure that there is only one iteration.

Reply:
Having `once` at this position made it easier to test the daemon, as the thread would cause the daemon to continue to loop after the first run. I'm realising now that `once` no longer needs to be a parameter in `declare_suspicious_replicas_bad`, so I shall remove it from there.

Reply:
I’m not sure I understand what you mean. The flow is `run()`→`declare_suspicious_replicas_bad()`→`run_daemon()`→`run_once()`. If `run_daemon` has `once=True`, then there should be a single execution of `run_once()`.